lib/data/external/GCP/GcpManagedUpload.js

```diff
@@ -202,33 +178,35 @@ class GcpManagedUpload {
      * @return {undefined}
      */
     send(callback) {
+        if (this.called || this.callback) {
+            return undefined;
+        }
         this.failed = false;
-        if (this.callback) return;
-        this.callback = callback || function newCallback(err) {
-            if (err) throw err;
-        };
-        if (this.error) {
-            this.callback(this.error);
-        } else {
-            if (this.totalBytes <= this.partSize) {
-                this.uploadSingle();
-            } else if (this.slicedParts) {
-                this.uploadBufferSlices();
-            } else if (this.body instanceof stream) {
-                // stream type
-                this.body.on('error', err => this.cleanUp(err))
-                .on('readable', () => this.chunkStream())
-                .on('end', () => {
-                    this.isDoneChunking = true;
-                    this.chunkStream();
+        this.called = true;
+        this.callback = callback;
+        assert.strictEqual(typeof callback, 'function',
+            new Error('Missing callback function for GcpManagedUpload'));
```

Comment:
We shouldn't need to instantiate a new `Error` here because it will be thrown as an `AssertionError` anyway.